### ISSUE TITLE
TransferUtility pause/resume is not working properly

### ISSUE THREAD

**Issue by chrisscholly** [aws/aws-sdk-ios#816]
`AWSS3TransferUtility` pause/resume feature is not working properly.

1. Upload a file (mine is about 2.5Mb for test purpose) using `AWSS3TransferUtility`.
2. Pause the upload (by invoking `AWSS3TransferUtilityUploadTask` `suspend()` method).
3. **Wait between 30-60 seconds.**
4. Resume the upload (by invoking `AWSS3TransferUtilityUploadTask` `resume()` method).
5. The upload does not resume. Either the upload start from scratch, or you have an error from the SDK (`The operation could not be completed. (com.amazonaws.AWSS3TransferUtilityErrorDomain error 2.)`)

Wait between 30-60 seconds is important because it works well when pausing/resuming instantly.
Here's the code to reproduce the issue :
```swift
import UIKit
import AWSS3

class ViewController: UIViewController {

    @IBOutlet weak var progressView: UIProgressView!
    @IBOutlet weak var infoLabel: UILabel!

    private var progressBlock: AWSS3TransferUtilityProgressBlock?
    private var completionBlock: AWSS3TransferUtilityUploadCompletionHandlerBlock?
    private var uploadTask: AWSS3TransferUtilityUploadTask?

    private lazy var transferUtility: AWSS3TransferUtility = {
        return AWSS3TransferUtility.default()
    }()

    override func viewDidLoad() {
        super.viewDidLoad()

        progressBlock = { [weak self] task, progress in
            guard let strongSelf = self else { return }
            DispatchQueue.main.async {
                strongSelf.infoLabel.text = "Uploading..."
                strongSelf.progressView.progress = Float(progress.fractionCompleted)
            }

        }

        completionBlock = { [weak self] task, error in
            guard let strongSelf = self else { return }
            if let error = error {
                DispatchQueue.main.async {
                    strongSelf.infoLabel.text = "Error: \(error.localizedDescription)"
                }
                return
            }
            DispatchQueue.main.async {
                strongSelf.infoLabel.text = "File Successfully Uploaded!"
            }
        }

        transferUtility.enumerateToAssignBlocks(forUploadTask: { [weak self] task, progressRef, completionRef in
            guard let strongSelf = self else { return }

            let progressPointer = AutoreleasingUnsafeMutablePointer<AWSS3TransferUtilityProgressBlock?>(&strongSelf.progressBlock)
            let completionPointer = AutoreleasingUnsafeMutablePointer<AWSS3TransferUtilityUploadCompletionHandlerBlock?>(&strongSelf.completionBlock)

            progressRef?.pointee = progressPointer.pointee
            completionRef?.pointee = completionPointer.pointee
        }, downloadTask: nil)
    }

    @IBAction func uploadButtonAction() {
        guard let url = Bundle.main.url(forResource: "test-file", withExtension: "mp4") else {
            return
        }
        upload(file: url)
    }

    @IBAction func pauseButtonAction() {
        uploadTask?.suspend()
        infoLabel.text = "Paused"
    }

    @IBAction func resumeButtonAction() {
        uploadTask?.resume()
    }

    private func upload(file url: URL) {
        let bucket = "blablabla-test-bucket"
        let key = "test-file.mp4"
        let contentType = "video/mp4"
        let expression = AWSS3TransferUtilityUploadExpression()
        expression.progressBlock = progressBlock
        

        let task = transferUtility.uploadFile(url,
                                              bucket: bucket,
                                              key: key,
                                              contentType: contentType,
                                              expression: expression,
                                              completionHandler: completionBlock)

        task.continueWith { (task) -> Any? in
            if let error = task.error {
                DispatchQueue.main.async {
                    self.infoLabel.text = "Error: \(error.localizedDescription)"
                }
                return nil
            }
            if let uploadTask = task.result {
                self.uploadTask = uploadTask
                DispatchQueue.main.async {
                    self.infoLabel.text = "Generating Upload File"
                }
            }
            return nil
        }
    }
}
```

* What service are you using?
`AWSS3TransferUtility`
* In what version of SDK are you facing the problem?
`2.6.10`
* Is the issue limited to Simulators / Actual Devices?
Both
* Can your problem be resolved if you bump to a higher version of SDK?
Nope
* Is this problem related to specific iOS version?
Only tested on iOS 11
* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
CocoaPods

Am I doing something wrong?

**Comment by kvasukib** [comment]
Hi Chris,

Thanks for reporting to us with a detailed description of the issue. We are working towards reproducing the issue.

**Comment by ThiyagarajanShivSankaran** [comment]
@kvasukib I am also facing the same problem.

**Comment by cbommas** [comment]
hello @ThiyagarajanShivSankaran, @chrisscholly 

We have addressed this issue in version 2.6.13 of the SDK. The issue is that the TransferUtility gets a `RequestTimeout` 400 error from S3 when the paused transfer is resumed. We have addressed this by adding error-retry capability for the TransferUtility.  You can adjust the number of times the TransferUtility retries by setting the retryLimit parameter. 

```
//Setup the transfer utility configuration
let tuConf = AWSS3TransferUtilityConfiguration()
tuConf.retryLimit = 5

//Register a transfer utility object
AWSS3TransferUtility.register(
    with: configuration!,
    transferUtilityConfiguration: tuConf,
    forKey: "transfer-utility-with-advanced-options"
)

//Look up the transfer utility object from the registry to use for your transfers.
let transferUtility = AWSS3TransferUtility.s3TransferUtility(forKey: "transfer-utility-with-advanced-options")
```
Please try this and let us know if you are still facing issues.
